src/cli/cli_tcp.cpp

```diff
@@ -0,0 +1,522 @@
+
+        SuccessOrExit(error = otTcpListenerInitialize(mInterpreter.mInstance, &mListener, &listenerArgs));
+        mListenerInitialized = true;
+    }
+
+exit:
+    return error;
+}
+
+otError TcpExample::ProcessDeinit(Arg aArgs[])
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aArgs[0].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(mEndpointInitialized || mListenerInitialized, error = OT_ERROR_INVALID_STATE);
+
+    if (mEndpointInitialized)
+    {
+        SuccessOrExit(error = otTcpEndpointDeinitialize(&mEndpoint));
+        mSendBusy            = false;
+        mEndpointInitialized = false;
+    }
+
+    if (mListenerInitialized)
+    {
+        SuccessOrExit(error = otTcpListenerDeinitialize(&mListener));
+        mListenerInitialized = false;
+    }
+
+exit:
+    return error;
+}
+
+otError TcpExample::ProcessBind(Arg aArgs[])
+{
+    otError    error;
+    otSockAddr sockaddr;
+
+    VerifyOrExit(mEndpointInitialized, error = OT_ERROR_INVALID_STATE);
+
+    SuccessOrExit(error = aArgs[0].ParseAsIp6Address(sockaddr.mAddress));
+    SuccessOrExit(error = aArgs[1].ParseAsUint16(sockaddr.mPort));
+    VerifyOrExit(aArgs[2].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
+
+    error = otTcpBind(&mEndpoint, &sockaddr);
+
+exit:
+    return error;
+}
+
+otError TcpExample::ProcessConnect(Arg aArgs[])
+{
+    otError    error;
+    otSockAddr sockaddr;
+
+    VerifyOrExit(mEndpointInitialized, error = OT_ERROR_INVALID_STATE);
+
+    SuccessOrExit(error = aArgs[0].ParseAsIp6Address(sockaddr.mAddress));
+    SuccessOrExit(error = aArgs[1].ParseAsUint16(sockaddr.mPort));
+    VerifyOrExit(aArgs[2].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
+
+    SuccessOrExit(error = otTcpConnect(&mEndpoint, &sockaddr, OT_TCP_CONNECT_NO_FAST_OPEN));
+    mEndpointConnected = false;
+
+exit:
+    return error;
+}
+
+otError TcpExample::ProcessSend(Arg aArgs[])
+{
+    otError error;
+
+    VerifyOrExit(mEndpointInitialized, error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(!mSendBusy, error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(mBenchmarkBytesTotal == 0, error = OT_ERROR_INVALID_STATE);
+
+    mSendLink.mNext = nullptr;
+    mSendLink.mData = mSendBuffer;
+    VerifyOrExit(!aArgs[0].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
+    mSendLink.mLength = OT_MIN(aArgs[0].GetLength(), sizeof(mSendBuffer));
+    memcpy(mSendBuffer, aArgs[0].GetCString(), mSendLink.mLength);
+    VerifyOrExit(aArgs[1].IsEmpty(), error = OT_ERROR_INVALID_ARGS);
+
+    SuccessOrExit(error = otTcpSendByReference(&mEndpoint, &mSendLink, 0));
+    mSendBusy = true;
+
+exit:
+    return error;
+}
+
+otError TcpExample::ProcessBenchmark(Arg aArgs[])
+{
+    otError  error;
+    uint32_t toSendOut;
+
+    VerifyOrExit(!mSendBusy, error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(mBenchmarkBytesTotal == 0, error = OT_ERROR_INVALID_STATE);
+
+    if (aArgs[0].IsEmpty())
+    {
+        mBenchmarkBytesTotal = 72 << 10;
```

Comment:
Just curious why this number? Should we defined as a constant?

Reply:
I chose it to be large enough to get a consistent measurement, but not so large that the benchmark would take more than a few seconds. I will define this as a constant.